### PR TITLE
Remove blacklist index from VecIndex

### LINKED ISSUE #4242: Query hang up at unittest case
**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
The process running milvus unit-test cases sometime could hang up at case "DBTest.QueryTest". The log output is like:
```
[ RUN      ] DBTest.QueryTest
[140447206609344-2020-11-17 08:39:16,514-DEBUG]: [ENGINE][Build][test_db] MetaUri: mock://:@:/
 (MetaFactory.cpp:49)
2020-11-17 08:39:16,514 INFO [default] [ENGINE][Build][test_db] Using Mock. Should only be used in test environment
[140447025846016-2020-11-17 08:39:16,524-DEBUG]: [ENGINE][GetHolder][test_db] Snapshots::GetHolderNoLock: Specified snapshot holder for collection "c1" not found (Snapshots.cpp:158)
[140447025846016-2020-11-17 08:39:16,527-DEBUG]: [ENGINE][Done][test_db] <CC(CID=0,CNAME="c1",N_CC=1194,N_CID=28,N_CNAME="c1",N_PC=1120,N_PID=268,N_PNAME="_default",LSN=1)> | DONE | OK | 2.228 ms (Operations.cpp:118)
[140447206609344-2020-11-17 08:39:16,532-DEBUG]: [SERVER][Insert][test_db] Auto generate entities id (DBImpl.cpp:556)
[140447206609344-2020-11-17 08:39:16,532-DEBUG]: [ENGINE][Insert][test_db] Insert entities into mem manager (DBImpl.cpp:582)
[140447206609344-2020-11-17 08:39:16,532-DEBUG]: [ENGINE][Flush][test_db] Begin flush all collections (DBImpl.cpp:793)
[140447206609344-2020-11-17 08:39:16,532-DEBUG]: [ENGINE][InternalFlush][test_db] Flushing collection: 28 (MemManagerImpl.cpp:183)
[140447206609344-2020-11-17 08:39:16,532-DEBUG]: [SERVER][PrintTimeRecord][test_db] MemCollection::Serialize collection 28: ApplyDeleteToFile (0.000001 second [0.000805 ms]) (TimeRecorder.cpp:47)
2020-11-17 08:39:16,533 WARNING [default] [ENGINE][OnApplyTimeoutCallback][test_db] N6milvus6engine8snapshot20SegmentFileOperationE Timeout! Try 1
2020-11-17 08:39:16,535 WARNING [default] [ENGINE][OnApplyTimeoutCallback][test_db] N6milvus6engine8snapshot20SegmentFileOperationE Timeout! Try 2
[140447206609344-2020-11-17 08:39:16,537-DEBUG]: [SERVER][PrintTimeRecord][test_db] BlockFormat::Write:/tmp/milvus_ss/db/collections/C_28/P_268/S_630/F_923: totally cost (0.000330 second [0.329870 ms]) (TimeRecorder.cpp:47)
[140447206609344-2020-11-17 08:39:16,537-DEBUG]: [ENGINE][WriteFields][test_db] Serialize raw file size: 84100 (SegmentWriter.cpp:121)
[140447206609344-2020-11-17 08:39:16,537-DEBUG]: [SERVER][PrintTimeRecord][test_db] BlockFormat::Write:/tmp/milvus_ss/db/collections/C_28/P_268/S_630/F_924: totally cost (0.000618 second [0.618357 ms]) (TimeRecorder.cpp:47)
[140447206609344-2020-11-17 08:39:16,537-DEBUG]: [ENGINE][WriteFields][test_db] Serialize raw file size: 404100 (SegmentWriter.cpp:121)
[140447206609344-2020-11-17 08:39:16,538-DEBUG]: [SERVER][PrintTimeRecord][test_db] BlockFormat::Write:/tmp/milvus_ss/db/collections/C_28/P_268/S_630/F_926: totally cost (0.000326 second [0.325998 ms]) (TimeRecorder.cpp:47)
[140447206609344-2020-11-17 08:39:16,538-DEBUG]: [ENGINE][WriteFields][test_db] Serialize raw file size: 84100 (SegmentWriter.cpp:121)
[140447206609344-2020-11-17 08:39:16,538-DEBUG]: [SERVER][PrintTimeRecord][test_db] SegmentWriter::WriteFields: totally cost (0.001596 second [1.596301 ms]) (TimeRecorder.cpp:47)
[140447206609344-2020-11-17 08:39:16,538-DEBUG]: [ENGINE][WriteDeletedDocs][test_db] Serialize deleted docs file size: 0 (SegmentWriter.cpp:198)
[140447206609344-2020-11-17 08:39:16,542-DEBUG]: [SERVER][PrintTimeRecord][test_db] SegmentWriter::WriteBloomFilter: Initialize bloom filter (0.003670 second [3.670410 ms]) (TimeRecorder.cpp:47)
[140447206609344-2020-11-17 08:39:16,542-DEBUG]: [SERVER][PrintTimeRecord][test_db] IdBloomFilterFormat::Write:/tmp/milvus_ss/db/collections/C_28/P_268/S_630/F_931.blf: totally cost (0.000123 second [0.123350 ms]) (TimeRecorder.cpp:47)
[140447206609344-2020-11-17 08:39:16,542-DEBUG]: [ENGINE][WriteBloomFilter][test_db] Serialize bloom filter file size: 60848 (SegmentWriter.cpp:163)
[140447206609344-2020-11-17 08:39:16,542-DEBUG]: [ENGINE][Serialize][test_db] New segment 630 of collection 28 committed (MemSegment.cpp:140)
[140447025846016-2020-11-17 08:39:16,545-DEBUG]: [ENGINE][SnapshotGCCallback][test_db] Snapshot 1194 ref_count = 3 To be removed (Snapshots.cpp:216)
[140447025846016-2020-11-17 08:39:16,545-DEBUG]: [ENGINE][Done][test_db] <MS(SS=1194,N_CC=1196,LSN=1)> | DONE | OK | 1.79 ms (Operations.cpp:118)
[140447206609344-2020-11-17 08:39:16,545-DEBUG]: [SERVER][PrintTimeRecord][test_db] MemCollection::Serialize collection 28: SerializeSegments (0.012448 second [12.447780 ms]) (TimeRecorder.cpp:47)
[140447206609344-2020-11-17 08:39:16,545-DEBUG]: [ENGINE][InternalFlush][test_db] Flushed collection: 28 (MemManagerImpl.cpp:189)
[140447206609344-2020-11-17 08:39:16,545-DEBUG]: [ENGINE][Flush][test_db] End flush all collections (DBImpl.cpp:795)
{"int64":{"values":[0,1,2,3,4]}}
[140447206609344-2020-11-17 08:39:16,545-DEBUG]: [SERVER][PrintTimeRecord][query] DBImpl::Query: segments to search: 1 (0.000011 second [0.011168 ms]) (TimeRecorder.cpp:47)
[140447206609344-2020-11-17 08:39:16,545-DEBUG]: [SERVER][print][query] [CACHE CPU] [item count]: 0, [usage] 0MB, [capacity] 256MB (Cache.inl:107)
[140446621751040-2020-11-17 08:39:16,545-DEBUG]: [SERVER][PrintTimeRecord][jobmgr_thread] JobMgr: build_tasks (0.000090 second [0.090495 ms]) (TimeRecorder.cpp:47)
[140446621751040-2020-11-17 08:39:16,545-DEBUG]: [SERVER][PrintTimeRecord][jobmgr_thread] JobMgr: optimize (0.000114 second [0.114425 ms]) (TimeRecorder.cpp:47)
[140446621751040-2020-11-17 08:39:16,545-DEBUG]: [SERVER][PrintTimeRecord][jobmgr_thread] JobMgr: calculate path (0.000107 second [0.107023 ms]) (TimeRecorder.cpp:47)
[140447000667904-2020-11-17 08:39:16,547-DEBUG]: [SERVER][PrintTimeRecord][taskloader_th] BlockFormat::Read:/tmp/milvus_ss/db/collections/C_28/P_268/S_630/F_923: totally cost (0.001432 second [1.431817 ms]) (TimeRecorder.cpp:47)
[140447000667904-2020-11-17 08:39:16,547-DEBUG]: [SERVER][insert_internal][taskloader_th] [CACHE CPU] Insert /tmp/milvus_ss/db/collections/C_28/P_268/S_630/F_923 size: 0MB into cache (Cache.inl:138)
[140447000667904-2020-11-17 08:39:16,547-DEBUG]: [SERVER][insert_internal][taskloader_th] [CACHE CPU] Count: 1, Usage: 0MB, Capacity: 256MB (Cache.inl:139)
[140447000667904-2020-11-17 08:39:16,547-DEBUG]: [SERVER][PrintTimeRecord][taskloader_th] SegmentReader::LoadField: _id: totally cost (0.001634 second [1.634017 ms]) (TimeRecorder.cpp:47)
[140447000667904-2020-11-17 08:39:16,548-DEBUG]: [SERVER][PrintTimeRecord][taskloader_th] SegmentReader::LoadVectorIndex: float_vector: prepare (0.001690 second [1.690382 ms]) (TimeRecorder.cpp:47)
[140447000667904-2020-11-17 08:39:16,549-DEBUG]: [SERVER][PrintTimeRecord][taskloader_th] BlockFormat::Read:/tmp/milvus_ss/db/collections/C_28/P_268/S_630/F_924: totally cost (0.001844 second [1.844449 ms]) (TimeRecorder.cpp:47)
[140447000667904-2020-11-17 08:39:16,549-DEBUG]: [SERVER][PrintTimeRecord][taskloader_th] SegmentReader::LoadField: float_vector: totally cost (0.001922 second [1.922389 ms]) (TimeRecorder.cpp:47)
[140447000667904-2020-11-17 08:39:16,550-DEBUG]: [SERVER][PrintTimeRecord][taskloader_th] SegmentReader::LoadField: _id: totally cost (0.000001 second [0.000690 ms]) (TimeRecorder.cpp:47)
[140447000667904-2020-11-17 08:39:16,550-DEBUG]: [SERVER][PrintTimeRecord][taskloader_th] SegmentReader::LoadUids copy uids: totally cost (0.000257 second [0.256894 ms]) (TimeRecorder.cpp:47)
[140447000667904-2020-11-17 08:39:16,551-DEBUG]: [SERVER][insert_internal][taskloader_th] [CACHE CPU] Insert /tmp/milvus_ss/db/collections/C_28/P_268/S_630/float_vector.tmp.index size: 0MB into cache (Cache.inl:138)
[140447000667904-2020-11-17 08:39:16,551-DEBUG]: [SERVER][insert_internal][taskloader_th] [CACHE CPU] Count: 2, Usage: 0MB, Capacity: 256MB (Cache.inl:139)
[140447000667904-2020-11-17 08:39:16,551-DEBUG]: [SERVER][PrintTimeRecord][taskloader_th] SegmentReader::LoadVectorIndex: float_vector: construct temp IDMAP index (0.003322 second [3.322234 ms]) (TimeRecorder.cpp:47)
[140447000667904-2020-11-17 08:39:16,552-DEBUG]: [SERVER][PrintTimeRecord][taskloader_th] BlockFormat::Read:/tmp/milvus_ss/db/collections/C_28/P_268/S_630/F_926: totally cost (0.001342 second [1.341637 ms]) (TimeRecorder.cpp:47)
[140447000667904-2020-11-17 08:39:16,552-DEBUG]: [SERVER][PrintTimeRecord][taskloader_th] SegmentReader::LoadField: int64: totally cost (0.001420 second [1.419821 ms]) (TimeRecorder.cpp:47)
[140447000667904-2020-11-17 08:39:16,553-DEBUG]: [SERVER][insert_internal][taskloader_th] [CACHE CPU] Insert /tmp/milvus_ss/db/collections/C_28/P_268/S_630/int64.tmp.index size: 0MB into cache (Cache.inl:138)
[140447000667904-2020-11-17 08:39:16,553-DEBUG]: [SERVER][insert_internal][taskloader_th] [CACHE CPU] Count: 3, Usage: 0MB, Capacity: 256MB (Cache.inl:139)
[140447000667904-2020-11-17 08:39:16,553-DEBUG]: [SERVER][PrintTimeRecord][taskloader_th] SegmentReader::LoadStructuredIndex: int64: create temp index (0.002054 second [2.054481 ms]) (TimeRecorder.cpp:47)
[140447000667904-2020-11-17 08:39:16,553-DEBUG]: [SERVER][PrintTimeRecord][taskloader_th] ExecutionEngineImpl::Load: totally cost (0.007181 second [7.181300 ms]) (TimeRecorder.cpp:47)
[140447000667904-2020-11-17 08:39:16,553-DEBUG]: [SERVER][PrintTimeRecord][taskloader_th] SearchTask::OnLoad 630: Search task load segment id: 630 DISK2CPU totally cost (0.007235 second [7.234763 ms]) (TimeRecorder.cpp:47)
[140446983882496-2020-11-17 08:39:16,553-DEBUG]: [SERVER][PrintTimeRecord][taskexecutor_th] [search][0] ExecutionEngineImpl::Search: Scalar field filtering (0.000028 second [0.028329 ms]) (TimeRecorder.cpp:47)
[140446983882496-2020-11-17 08:39:16,553-DEBUG]: [SERVER][PrintTimeRecord][taskexecutor_th] SegmentReader::LoadDeletedDocs: totally cost (0.000028 second [0.028249 ms]) (TimeRecorder.cpp:47)
[140446983882496-2020-11-17 08:39:16,554-DEBUG]: [SERVER][PrintTimeRecord][taskexecutor_th] [search][0] ExecutionEngineImpl::VecSearch: query prepare (0.000027 second [0.027166 ms]) (TimeRecorder.cpp:47)
[140446983882496-2020-11-17 08:39:16,573-DEBUG]: [SERVER][PrintTimeRecord][taskexecutor_th] [search][0] ExecutionEngineImpl::VecSearch: done (0.019253 second [19.253386 ms]) (TimeRecorder.cpp:47)
[140446983882496-2020-11-17 08:39:16,573-DEBUG]: [SERVER][PrintTimeRecord][taskexecutor_th] [search][0] ExecutionEngineImpl::Search: done (0.019543 second [19.543332 ms]) (TimeRecorder.cpp:47)
[140446983882496-2020-11-17 08:39:16,573-DEBUG]: [SERVER][PrintTimeRecord][taskexecutor_th] [search][0] DoSearch file id:630: search done (0.019583 second [19.583147 ms]) (TimeRecorder.cpp:47)
[140446983882496-2020-11-17 08:39:16,573-DEBUG]: [SERVER][PrintTimeRecord][taskexecutor_th] [search][0] DoSearch file id:630: reduce topk done (0.000038 second [0.037879 ms]) (TimeRecorder.cpp:47)
[140446983882496-2020-11-17 08:39:16,573-DEBUG]: [SERVER][PrintTimeRecord][taskexecutor_th] [search][0] DoSearch file id:630: totally cost (0.019648 second [19.648230 ms]) (TimeRecorder.cpp:47)
[140447206609344-2020-11-17 08:39:16,573-DEBUG]: [SERVER][print][query] [CACHE CPU] [item count]: 3, [usage] 0MB, [capacity] 256MB (Cache.inl:107)
[140447206609344-2020-11-17 08:39:16,573-DEBUG]: [SERVER][PrintTimeRecord][query] DBImpl::Query: execute query (0.028041 second [28.041212 ms]) (TimeRecorder.cpp:47)
```

I tried add more log like:
```
TEST_F(DBTest, QueryTest) {
         ...............
         ...............
    tr.RecordSection("Start query");    // Add log here
    status = db_->Query(ctx1, query_ptr, result);
    tr.RecordSection("Query done");
    ASSERT_TRUE(status.ok());        // Add log here
    ASSERT_EQ(result->row_num_, nq);
}
```

```
Status
DBImpl::Query(const server::ContextPtr& context, const query::QueryPtr& query_ptr, engine::QueryResultPtr& result) {
         .......................
         .......................

    // step 4: get entities by result ids
    std::vector<bool> valid_row;
    if (!query_ptr->field_names.empty()) {
        STATUS_CHECK(GetEntityByID(query_ptr->collection_id, result->result_ids_, query_ptr->field_names, valid_row,
                                   result->data_chunk_));
        rc.RecordSection("get entities");
    }

    rc.RecordSection("query finish");    // Add log here
    return Status::OK();
}
```
And the log output is:
```
TimeRecorderAuto::~TimeRecorderAuto: Start destruct ...
SegmentReader::LoadDeletedDocs: totally cost (0.000010 second [0.009588 ms])
[search][0] ExecutionEngineImpl::VecSearch: query prepare (0.000009 second [0.009427 ms])
[search][0] ExecutionEngineImpl::VecSearch: done (0.016000 second [16.000333 ms])
[search][0] ExecutionEngineImpl::Search: done (0.016057 second [16.057478 ms])
[search][0] DoSearch file id:677: search done (0.016064 second [16.063901 ms])
[search][0] DoSearch file id:677: reduce topk done (0.000009 second [0.008756 ms])
[search][0] DoSearch file id:677: totally cost (0.016078 second [16.077576 ms])
[140068828436864-2020-11-17 17:11:12,604-DEBUG]: [SERVER][print][query] [CACHE CPU] [item count]: 3, [usage] 0MB, [capacity] 256MB (Cache.inl:107)
DBImpl::Query: execute query (0.018586 second [18.586276 ms])
DBImpl::Query: query finish (0.000005 second [0.005059 ms])
```

From the above, we can see that method Query() finished but not exit.
**Steps/Code to reproduce behavior**

**Expected behavior**

**Method of installation**
- [ ] Docker/cpu
- [ ] Docker/gpu
- [x] Build from source

**Environment details**
- Hardware/Softwars conditions (OS, CPU, GPU, Memory)

- Milvus version: Branch 0.11.1

**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by BossZou**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

Fixes #4242

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Review by shengjun1985 — DISMISSED** [review]

**Review by jeffoverflow — APPROVED** [review]

**Review by shengjun1985 — APPROVED** [review]

**Review by wxyucs — APPROVED** [review]

File: ci/scripts/run_unittest.sh [hunk 1]
```diff
@@ -98,6 +98,10 @@ fi
 
 # run unittest
 for test in `ls ${UNITTEST_DIR}`; do
+    if [[ ${test} == *".log" ]] || [[ ${test} == *".info" ]]; then
```
Comment (cydrain) [inline ci/scripts/run_unittest.sh:101]: why need this ?

File: core/coverage.sh [hunk 2]
```diff
@@ -27,6 +27,10 @@ fi
 
 # run unittest
 for test in `ls ${UNITTEST_DIR}`; do
+    if [[ ${test} == *".log" ]] || [[ ${test} == *".info" ]]; then
```
Comment (cydrain) [inline core/coverage.sh:30]: why need this check ?

File: core/src/db/Types.h [hunk 3]
```diff
@@ -160,7 +160,7 @@ struct AttrsData {
 
 ///////////////////////////////////////////////////////////////////////////////////////////////////
 struct QueryResult {
-    uint64_t row_num_;
+    uint64_t row_num_ = 0;
```
Comment (cydrain) [inline core/src/db/Types.h:163]: Can we use "int64_t" instead of "uint64_t" ?

File: core/src/query/GeneralQuery.h [hunk 4]
```diff
@@ -70,18 +70,18 @@ struct RangeQuery {
 using RangeQueryPtr = std::shared_ptr<RangeQuery>;
 
 struct VectorRecord {
-    size_t vector_count;
+    size_t vector_count = 0;
```
Comment (cydrain) [inline core/src/query/GeneralQuery.h:73]: suggest to use "int64_t" instead

File: core/src/utils/Error.h [hunk 5]
```diff
@@ -112,6 +112,7 @@ constexpr ErrorCode DB_BLOOM_FILTER_ERROR = ToDbErrorCode(9);
 constexpr ErrorCode DB_PARTITION_NOT_FOUND = ToDbErrorCode(10);
 constexpr ErrorCode DB_OUT_OF_STORAGE = ToDbErrorCode(11);
 constexpr ErrorCode DB_META_QUERY_FAILED = ToDbErrorCode(12);
+constexpr ErrorCode DB_FILE_NOT_FOUND = ToDbErrorCode(13);
```
Comment (cydrain) [inline core/src/utils/Error.h:115]: seems duplicate with DB_NOT_FOUND or DB_INVALID_PATH

**Review by yhmo — APPROVED** [review]

File: ci/scripts/run_unittest.sh (hunk 1 above)
Comment (BossZou) [reply ci/scripts/run_unittest.sh:101]: For some reason there may be log files under unittest foler

**Pull request merged**